### PR TITLE
Remove special render build

### PR DESCRIPTION
**Pull request by firstdoit**
#### What is the purpose of this pull request?
Removes all special build steps from the `render` service on `watch` and `publish`. Now every folder in the app is uploaded normally, with every file as-is.

Note that `eslint` was lost in the process. We need to think of a way to allow services to configure their own checks (via `manifest.json` configuration, maybe?).

#### What problem is this solving?
The fact that we are changing our approach in how the render apps in a workspace work makes any build step unnecessary. Instead of individual builds, all apps in a workspace will be compiled by a single webpack process. Adieu, SystemJS.

#### How should this be manually tested?
Thoroughly. 😝 
But really just notice that there's no more `.build/` folder upon watching and publishing.

#### Screenshots or example usage
```
guilherme@trinsic ~/P/V/hello-world (master)> releasy patch stable
Release: patch
cat: no such file or directory: package.json
package.json not found, switching to manifest.json
Old version: 0.1.18
New version: 0.1.19
prompt: Are you sure?:  (yes)
Starting release...
Version bumped to 0.1.19
File manifest.json added > git add manifest.json
File manifest.json committed > git commit manifest.json -m "Release v0.1.19"
Tag created: v0.1.19 > git tag v0.1.19 -m "Release v0.1.19"
Pushed commit and tags > git push && git push --tags
info:    Published app vtex.hello-world@0.1.19 successfully
Post releasy > vtex publish
All steps finished successfuly.
```

`curl "http://api.beta.vtex.com/pokestore/render-demo-aug-16/apps/vtex.hello-world@0.1.19/files"`

```
{
    "files": [
        {
            "path": "manifest.json",
            "hash": "\"F4B6C28F22883EB395F8A08C93D800AE\""
        },
        {
            "path": "render/index.js",
            "hash": "\"94195661B62E935900F2DA145681A710\""
        },
        {
            "path": "render/nested.js",
            "hash": "\"74E526A44E3AADAEDB17412216D2F3D8\""
        },
        {
            "path": "render/routes.json",
            "hash": "\"66A651DC7147B1094F4412A59EA3082D\""
        },
        {
            "path": "render/simple.js",
            "hash": "\"5195EF02CEE80BDCCFA2C7C7B46BB011\""
        }
    ],
    "nextMarker": null
}
```

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: src/file.js [hunk 1]
```diff
@@ -1,38 +1,19 @@
...
-  '!node_modules/**/*',
-]
-
-export function joinWithBuild (root) {
-  return path.join(root, '.build/')
-}
-
 export function listLocalFiles (root) {
-  return bbGlob('{.build/**/*,manifest.json}', {
+  return bbGlob('{*/**,manifest.json}', {
```
Comment (tamorim) [inline src/file.js]: @firstdoit This glob pattern only looks for folders on the root and the manifest file?

Comment (firstdoit) [inline src/file.js]: Yes. We don't want READMEs, gitignores, etc etc. All services live in folders.

**Comment by tamorim** [comment]
@firstdoit Packages you can remove from `package.json`:
```javascript
{    
    "gulp": "^3.9.1",
    "gulp-autoprefixer": "^3.1.0",
    "gulp-babel": "^6.1.2",
    "gulp-cached": "^1.1.0",
    "gulp-cssnano": "^2.1.2",
    "gulp-eslint": "^3.0.1",
    "gulp-filter": "^4.0.0",
    "gulp-if": "^2.0.1",
    "gulp-imagemin": "^3.0.2",
    "gulp-less": "^3.1.0",
    "gulp-remember": "^0.3.1",
    "gulp-sass": "^2.3.2",
    "gulp-sourcemaps": "^1.6.0",
    "gulp-tap": "^0.1.3",
    "gulp-uglify": "^1.5.4",
    "gulp-vtex-render": "^0.5.0",
    "gulp-watch": "^4.3.8",
    "babel-plugin-import-rename": "^1.0.1",
    "babel-plugin-transform-es2015-modules-systemjs": "^6.9.0",
    "babel-plugin-vtex-render-route": "^1.3.0",
    "babel-preset-react": "^6.5.0",
    "eslint-config-vtex-react": "^1.0.2",
    "eslint-plugin-react": "^5.0.1"
}
```

**Comment by tamorim** [comment]
@firstdoit Packages you can move from **dependencies** to **devDependencies**:
```javascript
{
    "babel-eslint": "^6.0.2",
    "babel-preset-es2015": "^6.13.0",
    // It's better to use stage 2 instead of stage 0
    "babel-preset-stage-2": "^6.11.0",
    "eslint": "^2.10.2",
    "eslint-config-vtex": "^4.0.0"
}
```

And we can add the `.eslinrc` on `.npmignore` again :)

File: src/file.js (hunk 1 above)
Comment (tamorim) [inline src/file.js]: I agree, it's just to ensure this behavior hehe :+1:

**Comment by tamorim** [comment]
@firstdoit Actually, we have to leave the `.eslintrc` and `eslint` stuff on **dependencies** because of the `vtex setup` command :|

**Comment by firstdoit** [comment]
Jaha, indeed!

**Comment by firstdoit** [comment]
Depends on https://github.com/vtex/apps-client-node/pull/4 and https://github.com/vtex/vbase-client-node/pull/1

File: src/modules/apps.js [hunk 2]
```diff
@@ -236,7 +236,8 @@ export default {
       log.debug('Starting to uninstall app', app)
       const appRegex = new RegExp(`^${vendorPattern}\.${namePattern}@${wildVersionPattern}$`)
       if (!appRegex.test(app)) {
-        return log.error('Invalid app format, please use <vendor>.<name>@<version>')
+        log.error('Invalid app format, please use <vendor>.<name>@<version>')
```
Comment (tamorim) [inline src/modules/apps.js]: @firstdoit If the uninstall doesn't need the version anymore, we should update the message, no?

**Comment by tamorim** [comment]
@firstdoit Remember to update the toolbelt usage for the `node-vtex-api`!

**Comment by firstdoit** [comment]
@tamorim done!

**Comment by tamorim** [comment]
@firstdoit LGTM :+1:

**Pull request merged**